### DISCUSSION TITLE
Future UI Framework

### DISCUSSION THREAD

**Discussion by taooceros** [Flow-Launcher/Flow.Launcher#1273] [Polls]
Let's give a vote for the UI framework we want to use for flow 2.0 in the future.

Pros of Avalonia:
1. Similar to WPF (except some style)
2. Capable for trimming (or future NativeAOT)
3. Better Performance (probably)
4. Potential Cross Platform

Pros of WinUI3 (MAUI):
1. Native Modern Control and style
2. Latest Framework
3. Better Performance (probably?)

Cons:
1. Old windows are depreciated
2. Some feature only available for win11

Pros of WPF:
1. We don't need to change anything

Cons:
1. A few more years may be needed for trimming
2. No native modern style (but @onesounds almost finish them)

**Comment by jjw24** [comment]
@taooceros Do you know in terms of performance and responsiveness, which one is better, Avalonia or WinUI3?

**Reply by taooceros** [reply]
I am not sure, but I do see some complain about the performance of winui3 is worse than uwp. Though, both are supposed to be better than wpf.

**Reply by jjw24** [reply]
Let's look into it and record the comparisons here.
